java/ql/src/experimental/Security/CWE/CWE-094/JShellInjection.qll

```diff
@@ -0,0 +1,44 @@
+import java
+import semmle.code.java.dataflow.FlowSources
+
+/** A sink for JShell expression injection vulnerabilities. */
+class JShellInjectionSink extends DataFlow::Node {
+  JShellInjectionSink() {
+    this.asExpr() = any(JShellEvalCall jsec).getArgument(0)
+    or
+    this.asExpr() = any(SourceCodeAnalysisWrappersCall scawc).getArgument(0)
+    or
+    exists(MethodAccess ma |
```

Comment:
Do this using `isAdditionalTaintStep`. `analyzeCompletion​` yields a tainted `SourceCodeAnalysis.CompletionInfo`; `remaining()` of a tainted `SourceCodeAnalysis.CompletionInfo` yields a tainted string, as does `.source()`; `eval(...)` is the sink as already stated above.

Reply:
ok.